draft-js-side-toolbar-plugin/src/components/Toolbar/index.js

```diff
@@ -50,33 +75,48 @@ export default class Toolbar extends React.Component {
         editorRoot = editorRoot.parentNode;
       }
 
+      const position = {
+        top: node.offsetTop + editorRoot.offsetTop,
+        transform: 'scale(1)',
+        transition: 'transform 0.15s cubic-bezier(.3,1.2,.2,1)',
+      };
+      // TODO: remove the hard code(width for the hover element)
+      if (this.props.position === 'right') {
+        // eslint-disable-next-line no-mixed-operators
+        position.left = editorRoot.offsetLeft + editorRoot.offsetWidth + 80 - 36;
+      } else {
+        position.left = editorRoot.offsetLeft - 80;
+      }
+
+
       this.setState({
-        position: {
-          top: node.offsetTop + editorRoot.offsetTop,
-          left: editorRoot.offsetLeft - 80,
-          transform: 'scale(1)',
-          transition: 'transform 0.15s cubic-bezier(.3,1.2,.2,1)',
-        },
+        position,
       });
     }, 0);
   }
 
   render() {
     const { theme, store } = this.props;
+
     return (
       <div
         className={theme.toolbarStyles.wrapper}
         style={this.state.position}
       >
-        {this.props.structure.map((Component, index) => (
-          <Component
-            key={index}
-            getEditorState={store.getItem('getEditorState')}
-            setEditorState={store.getItem('setEditorState')}
-            theme={theme}
-          />
-        ))}
+        <BlockTypeSelect
```

Comment:
@freedomlang @juliankrispel 

With this PR you've hardcoded this `BlockTypeSelect` into the component and so you can no longer provide your own button. This blocks an upgrade to the latest version of this plugin in my app since I'm using a custom icon. Have you considered that the render prop could be responsible for rendering the whole block type select if provided?

https://github.com/draft-js-plugins/draft-js-plugins/issues/897 discussed something similar.